sdk/clientcore/core/src/main/java/io/clientcore/core/credentials/oauth/OAuthTokenRequestProperties.java

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.credentials.oauth;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * <p>
+ * The {@link OAuthTokenRequestProperties} is a class used to provide additional information and context when requesting an
+ * access token from an authentication source. It allows you to customize the token request and specify additional
+ * parameters, such as scopes, claims, or authentication options.
+ * </p>
+ *
+ * <p>
+ * Here's a high-level overview of how you can use the {@link OAuthTokenRequestProperties}:
+ * </p>
+ *
+ * <ol>
+ * <li>Create an instance of the {@link OAuthTokenRequestProperties} class and configure the required properties.
+ * The {@link OAuthTokenRequestProperties} class allows you to specify the scopes or resources for which you want to request
+ * an access token, as well as any additional claims.</li>
+ *
+ * <li>Pass the OAuthTokenRequestProperties instance to the appropriate authentication client or mechanism when
+ * requesting an access token. The specific method or API to do this will depend on the authentication mechanism
+ * you are using.</li>
```

Comment:
I mean you only pass one of these to `getToken`, right? I'd remove/rework this so users aren't left wondering "what else do I do with this.."